### PR TITLE
fix Vivit for video classification example

### PR DESCRIPTION
**Pull request by Geometrein**
# What does this PR do?
Both Vivit for video classification example implementations were not working.
Some of the issues included:
- Missing or unnecessary library Imports
- Multiple variables referenced before assignment.
- Missing docstring for `sample_frame_indices()` method
- Logical error in `sample_frame_indices()` method that would cause Numpy ValueError.
- Missing or unnecessary lines of code.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by ArthurZucker** [comment]
cc @amyeroberts

File: src/transformers/models/vivit/modeling_vivit.py [hunk 1]
```diff
@@ -532,8 +532,17 @@ def forward(
...
+        ...     Args:
+        ...         clip_len (`int`): Total number of frames to sample
+        ...         frame_sample_rate (`int`): Frame sample rate
+        ...         seg_len (`int`):  Number of frames in a sample segment
+        ...     Returns:
+        ...         result (`List[int]`): List of sampled video frame indices
+        ...     '''
         ...     converted_len = int(clip_len * frame_sample_rate)
-        ...     end_idx = np.random.randint(converted_len, seg_len)
+        ...     end_idx = np.random.randint(converted_len, converted_len + seg_len)
```
Comment (amyeroberts) [inline src/transformers/models/vivit/modeling_vivit.py:545]: I don't think this is correct - we can't select a frame longer than the seqence length

File: src/transformers/models/vivit/modeling_vivit.py [hunk 2]
```diff
@@ -547,8 +556,8 @@ def forward(
         >>> container = av.open(file_path)
 
         >>> # sample 32 frames
-        >>> indices = sample_frame_indices(clip_len=32, frame_sample_rate=1, seg_len=len(videoreader))
-        >>> video = videoreader.get_batch(indices).asnumpy()
+        >>> indices = sample_frame_indices(clip_len=32, frame_sample_rate=2, seg_len=2)
```
Comment (amyeroberts) [inline src/transformers/models/vivit/modeling_vivit.py:559]: This also isn't correct. `seq_len` should be the total number of frames in the video segment

Comment (amyeroberts) [inline src/transformers/models/vivit/modeling_vivit.py:559]: Why update the frame_sample_rate here?

File: src/transformers/models/vivit/modeling_vivit.py [hunk 3]
```diff
@@ -532,8 +532,17 @@ def forward(
...
         >>> def sample_frame_indices(clip_len, frame_sample_rate, seg_len):
+        ...     '''
+        ...     Sample frames from the video.
+        ...     Args:
+        ...         clip_len (`int`): Total number of frames to sample
+        ...         frame_sample_rate (`int`): Frame sample rate
+        ...         seg_len (`int`):  Number of frames in a sample segment
+        ...     Returns:
+        ...         result (`List[int]`): List of sampled video frame indices
+        ...     '''
```
Comment (amyeroberts) [inline src/transformers/models/vivit/modeling_vivit.py:543]: To match library doc string formatting 

```suggestion
        ...     """
        ...     Sample frames from the video.
        ...     Args:
        ...         clip_len (`int`): 
        ...             Number of frames to sample
        ...         frame_sample_rate (`int`):
        ...             Frame sample rate
        ...         seg_len (`int`):
        ...             Total number of frames in a sample segment
        ...    
        ...     Returns:
        ...         indices (`List[int]`): List of sampled video frame indices
        ...     """
```

File: src/transformers/models/vivit/modeling_vivit.py [hunk 4]
```diff
@@ -532,8 +532,17 @@ def forward(
 
 
         >>> def sample_frame_indices(clip_len, frame_sample_rate, seg_len):
+        ...     '''
```
Comment (amyeroberts) [inline src/transformers/models/vivit/modeling_vivit.py:535]: Nice! Could you add these to the other doc string examples with `sample_frame_indices` too?

**Review by amyeroberts — COMMENTED** [review]
Hi @Geometrein, thanks for opening this PR!

The currently suggested updates don't match the intended logic for `sample_frame_indices` and some of the changes aren't needed. As it stands, the PR won't be merged in. 

The doc string is a useful addition and we should add this across the library to the other examples.

**Pull request closed without merging**